### PR TITLE
feat(web2): WPA2 Enterprise (802.1x) frontend support

### PR DESCRIPTION
**Pull request by GregoryIvo**
> **Note**: We are using the Conventional Commits convention for our pull request titles. Please take a look at the [PR title format document](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#submitting-the-changes) for the supported [types](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#type) and [scopes](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#scope).

Brief description of the PR. [e.g. Added `null` check on `object` to avoid `NullPointerException`]

**Related Issue:** This PR fixes/closes {issue number}

**Description of the solution adopted:** A more detailed description of the changes made to solve/close one or more issues. If the PR is simple and easy to understand this section can be skipped

**Screenshots:** If applicable, add screenshots to help explain your solution

**Manual Tests**: Optional description of the tests performed to check correct functioning of changes, useful for an efficient review

**Any side note on the changes made:** Description of any other change that has been made, which is not directly linked to the issue resolution [e.g. Code clean up/Sonar issue resolution]

### CODE REVIEW

**Comment by GregoryIvo** [comment]
GWT
- [x] tab creation
- [x] basic UI elements
- [x]  populate help fields
- [x] fix bug when storing settings in meta type
- [x] fix meta-type retrieval
- [x] hide and show tab when Enterprise is selected
- [x] hide enterprise in AP mode
- [x] add fields for certificates and support for other security types

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/NetworkTabsUi.java [hunk 1]
```diff
@@ -328,6 +346,7 @@ private void showWirelessTabs() {
         this.wirelessTabAnchorItem.setEnabled(true);
 
         insertTab(this.wirelessTabAnchorItem);
+        insertTab(this.net8021xTabAnchorItem);
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/NetworkTabsUi.java]: We should show the tab and insert the anchor item only if we are on generic profiles that have `NetworkManager`, i.e. only if `this.isNet2` is true. Look at the `updateTabs()` method.

**Review by marcellorinaldo — CHANGES_REQUESTED** [review]

File: kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties [hunk 2]
```diff
@@ -620,6 +620,16 @@ netWifiChannelMissingError=The chosen network uses a channel that is not allowed
 netWifiCountryCode=Country Code
 netWifiCountryCodeLabel=Country Code
 
+netWifiWireless8021x=802-1x
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties:623]: Remember to update also the japanese version of this file.

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/GwtWifiSecurity.java [hunk 3]
```diff
@@ -17,5 +17,6 @@ public enum GwtWifiSecurity {
     netWifiSecurityWEP,
     netWifiSecurityWPA,
     netWifiSecurityWPA2,
+    netWifiSecurityWPA2WPA3Enterprise,
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/GwtWifiSecurity.java:20]: Please update copyright header

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/Gwt8021xInnerAuth.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package org.eclipse.kura.web.shared.model;
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/Gwt8021xInnerAuth.java:13]: Please add copyright header.

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/Gwt8021xEap.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
+package org.eclipse.kura.web.shared.model;
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/Gwt8021xEap.java:13]: Please add copyright header.

**Comment by GregoryIvo** [comment]
closed in favor of: https://github.com/eclipse/kura/pull/4852

**Pull request closed without merging**